server/routers/api-router.js

```diff
@@ -141,9 +145,17 @@ router.get("/api/status-page/monitor-list", cache("5 minutes"), async (_request,
         await checkPublished();
         const publicGroupList = [];
         let list = await R.find("group", " public = 1 ORDER BY weight ");
-
         for (let groupBean of list) {
-            publicGroupList.push(await groupBean.toPublicJSON());
+            let monitorGroup = await groupBean.toPublicJSON()
+            if ((await getSettings("statusPage")).statusPageTags=="visible") {
+                monitorGroup.monitorList = await Promise.all(monitorGroup.monitorList.map( async (monitor)=>{
+                    // Includes tags as an array in response, allows for tags to be displayed on public status page
+                    let tags = await R.getAll("SELECT mt.monitor_id,mt.value, tag.name, tag.color FROM monitor_tag mt JOIN tag ON mt.tag_id = tag.id WHERE mt.monitor_id = ?", [monitor.id]);
+                    return {...monitor,tags: tags}
+                }))
+            }
```

Comment:
Logically it looks pretty sound, but is there any chance you can get this to format a little cleaner? It was a bit of a pain to read through. 

- the if clause on 150 I kept misinterpreting the parens
- the SQL could be formatted for clarity
- the callback chaining could likely do with some newlines and indenting

Also, wont this `getSettings` for every `groupBean`? I don't think that is necessary.

Reply:
> Also, wont this `getSettings` for every `groupBean`?

Good catch, I've cleaned up the formatting and loop in the latest commit so it should be quite a lot more readable.